aws-guide/AWS-GUIDE.md

```diff
@@ -0,0 +1,237 @@
+
+   ```shell
+   POLICY_ARN=$(aws iam create-policy --policy-name s3-bucket-access-policy --policy-document '{
+       "Version": "2012-10-17",
+       "Statement": [
+           {
+               "Sid": "VisualEditor0",
+               "Effect": "Allow",
+               "Action": [
+                   "s3:PutObject",
+                   "s3:GetObjectAcl",
+                   "s3:GetObject",
+                   "s3:ListBucket",
+                   "s3:DeleteObject",
+                   "s3:PutObjectAcl"
+               ],
+               "Resource": [
+                   "arn:aws:s3:::<your-bucket-name>",
+                   "arn:aws:s3:::<your-bucket-name>/*"
+               ]
+           }
+       ]
+   }' | jq -r .Policy.Arn)
+   ```
+
+2. Create an `nx-cloud-service-account` service account and an IAM role attached to it, using the Permission Policy you created above (replace the items in angled brackets):
+
+   ```shell
+   eksctl create iamserviceaccount \
+     --name nx-cloud-service-account \
+     --cluster nx-cloud-cluster \
+     --attach-policy-arn $POLICY_ARN \
+     --override-existing-serviceaccounts \
+     --approve \
+     --role-name nx-cloud-cluster-s3-access-role
+   ```
+
+3. Add these options to the helm.yaml file:
+
+   ```yaml
+   awsS3:
+      enabled: true
+      bucket: '<your-bucket-name>'
+      serviceAccountName: 'nx-cloud-service-account'
+      # accelerated: true  uncomment when using accelerated bucket
+      # endpoint: ''  uncomment when using a custom endpoint
+   ```
+
+4. Push the new config: `helm upgrade --install nx-cloud nx-cloud/nx-cloud --values=./helm-values.yml`
+
+## 7. Set-up external secrets management
+
+1. Make sure you remove any previous k8s secrets named `nx-cloud-k8s-secret`, otherwise it will affect the below
+   1. `kubectl delete secret nx-cloud-k8s-secret`
+2. [Create a new `AWS Secret Manager` secret](https://docs.aws.amazon.com/secretsmanager/latest/userguide/create_secret.html) with key/value pairs called `nx-cloud-secrets`
+   1. ![aws-secret-manager-example](/aws-guide/aws-secret-manager-example.png)
+3. Create a policy that can read secrets:
+
+```shell
+POLICY_ARN=$(aws iam create-policy --policy-name secrets-reader --policy-document '{
+    "Version": "2012-10-17",
+    "Statement": [
+     {
+            "Effect": "Allow",
+            "Action": [
+                "secretsmanager:GetResourcePolicy",
+                "secretsmanager:GetSecretValue",
+                "secretsmanager:DescribeSecret",
+                "secretsmanager:ListSecretVersionIds"
+            ],
+            "Resource": [
+                "<SECRET-ARN>"
+            ]
+        },
+        {
+            "Effect": "Allow",
+            "Action": "secretsmanager:ListSecrets",
+            "Resource": "*"
+        }
+    ]
+}
+' | jq -r .Policy.Arn)
```

Comment:
We probably want to add a note that people will need `jq` installed to follow along. I think that is easier than writing instructions on where to pull this value from manually.

Reply:
makes sense - I think people will notice when they get to this part that `jq` is used, and even then, they won't really need it, as they can just copy/paste the resulting ARN that appears in the output, so I might skip the extra note just for brevity's sake, as this guide is already chunky 😄